tasks/eva_2850/fix_discordant_variants.py

```diff
@@ -147,24 +161,35 @@ def correct_discordant_rs_and_insert_into_db(rs_variant, ss_records, assembly):
         return None, None
 
 
-def resolve_collision_and_insert_rs(rs_variant, rs_with_new_start, variant_in_db, assembly):
+def resolve_collision_and_insert_rs(rs_variant, rs_with_new_start, variant_in_dbsnp, variant_in_eva, assembly):
     dbsnp_cve_collection = mongo_source.mongo_handle[mongo_source.db_name][DBSNP_CLUSTERED_VARIANT_ENTITY]
     dbsnp_cvoe_collection = mongo_source.mongo_handle[mongo_source.db_name][DBSNP_CLUSTERED_VARIANT_OPERATION_ENTITY]
+    eva_cve_collection = mongo_source.mongo_handle[mongo_source.db_name][EVA_CLUSTERED_VARIANT_ENTITY]
+    eva_cvoe_collection = mongo_source.mongo_handle[mongo_source.db_name][EVA_CLUSTERED_VARIANT_OPERATION_ENTITY]
+
+    variant_in_db = variant_in_dbsnp if variant_in_dbsnp else variant_in_eva
 
     # For priority refer to:
     # https://github.com/EBIvariation/eva-accession/blob/0b2ae4cdb6f74152c5443c3831c02c1d76cf93f9/eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteredVariantMergingPolicy.java#L40
     if rs_with_new_start['accession'] < variant_in_db['accession']:
         logger.info(
             f"delete rs with wrong start and the one being merged: \nWrong start: {rs_variant} \nMerged: {variant_in_db}")
-        dbsnp_cve_collection.with_options(write_concern=WriteConcern("majority")) \
-            .delete_many({'_id': {'$in': [rs_variant['_id'], variant_in_db['_id']]}})
+        if variant_in_dbsnp:
+            dbsnp_cve_collection.with_options(write_concern=WriteConcern("majority")) \
+                .delete_many({'_id': {'$in': [rs_variant['_id'], variant_in_db['_id']]}})
+            merge_event = create_merge_event(variant_in_db, rs_with_new_start)
```

Comment:
suggest creating the event before the deletion. see rationale [here](https://github.com/EBIvariation/eva-accession/pull/352#discussion_r943633043)

Reply:
done